test/reencrypt.ts

```diff
@@ -1,18 +1,163 @@
+import {
+  EBOOL_T,
+  EBYTES64_T,
+  EBYTES128_T,
+  EBYTES256_T,
+  EUINT4_T,
+  EUINT8_T,
+  EUINT16_T,
+  EUINT32_T,
+  EUINT64_T,
+  EUINT128_T,
+  EUINT160_T,
+  EUINT256_T,
+  verifyType,
+} from "./handleTypeCheck";
 import { Signers } from "./signers";
 import { FhevmInstances } from "./types";
 
+export async function reencryptEbool(
+  signers: Signers,
+  instances: FhevmInstances,
+  user: string,
+  handle: bigint,
+  contractAddress: string,
+): Promise<bigint> {
+  verifyType(handle, EBOOL_T);
+  return reencryptHandle(signers, instances, user, handle, contractAddress);
+}
+
+export async function reencryptEuint4(
+  signers: Signers,
+  instances: FhevmInstances,
+  user: string,
+  handle: bigint,
+  contractAddress: string,
+): Promise<bigint> {
+  verifyType(handle, EUINT4_T);
+  return reencryptHandle(signers, instances, user, handle, contractAddress);
+}
+
+export async function reencryptEuint8(
+  signers: Signers,
+  instances: FhevmInstances,
+  user: string,
+  handle: bigint,
+  contractAddress: string,
+): Promise<bigint> {
+  verifyType(handle, EUINT8_T);
+  return reencryptHandle(signers, instances, user, handle, contractAddress);
+}
+
+export async function reencryptEuint16(
+  signers: Signers,
+  instances: FhevmInstances,
+  user: string,
+  handle: bigint,
+  contractAddress: string,
+): Promise<bigint> {
+  verifyType(handle, EUINT16_T);
+  return reencryptHandle(signers, instances, user, handle, contractAddress);
+}
+
+export async function reencryptEuint32(
+  signers: Signers,
+  instances: FhevmInstances,
+  user: string,
+  handle: bigint,
+  contractAddress: string,
+): Promise<bigint> {
+  verifyType(handle, EUINT32_T);
+  return reencryptHandle(signers, instances, user, handle, contractAddress);
+}
+
+export async function reencryptEuint64(
+  signers: Signers,
+  instances: FhevmInstances,
+  user: string,
+  handle: bigint,
+  contractAddress: string,
+): Promise<bigint> {
+  verifyType(handle, EUINT64_T);
+  return reencryptHandle(signers, instances, user, handle, contractAddress);
+}
+
+export async function reencryptEuint128(
+  signers: Signers,
+  instances: FhevmInstances,
+  user: string,
+  handle: bigint,
+  contractAddress: string,
+): Promise<bigint> {
+  verifyType(handle, EUINT128_T);
+  return reencryptHandle(signers, instances, user, handle, contractAddress);
+}
+
+export async function reencryptEuint160(
```

Comment:
Rename this function to `reencryptEaddress` and it should return a correctly formatted hexstring , like what was did for the decryptAddress